packages/read-model/src/Flag/FlagsProvider.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Flag;
+
+use Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade;
+
+class FlagsProvider
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade
+     */
+    protected $flagFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Flag\Flag[]
+     */
+    protected $allFlags;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade $flagFacade
+     */
+    public function __construct(FlagFacade $flagFacade)
+    {
+        $this->flagFacade = $flagFacade;
+    }
+
+    /**
+     * @param int[] $flagIds
+     * @return \Shopsys\FrameworkBundle\Model\Product\Flag\Flag[]
+     */
+    public function getFlagsByIds(array $flagIds): array
+    {
+        if ($this->allFlags === null) {
+            $this->allFlags = $this->flagFacade->getAll();
+        }
+
+        $flags = [];
```

Comment:
The provider itself is awesome, great job!

I was staring on the `foreach-if` for a while, and then I realized we just filter flags by something. So I've rewritten this statement to
```php
return array_filter($this->allFlags, function (Flag $flag) use ($flagIds) {
    return in_array($flag->getId(), $flagIds, true);
});
```

And then I realized how horrible filter/mapping/reduce operations looks in the PHP so... It's up to you - keep the original implementation and if you like the `filter`, change it :-)

Reply:
btw I didn't test it

Reply:
I will keep the original implementation as I believe it is easier to read.